src/main/kotlin/no/nav/syfo/personstatus/application/PersonTildelingService.kt

```diff
@@ -3,34 +3,57 @@ package no.nav.syfo.personstatus.application
 import no.nav.syfo.personstatus.domain.PersonIdent
 import no.nav.syfo.personstatus.domain.PersonOversiktStatus
 import no.nav.syfo.personstatus.domain.VeilederBrukerKnytning
+import no.nav.syfo.personstatus.infrastructure.clients.veileder.VeilederClient
+import org.slf4j.LoggerFactory
 import kotlin.also
 import kotlin.collections.forEach
 
 class PersonTildelingService(
     private val personoversiktStatusRepository: IPersonOversiktStatusRepository,
     private val personBehandlendeEnhetService: PersonBehandlendeEnhetService,
+    private val veilederClient: VeilederClient,
 ) {
     suspend fun lagreKnytningMellomVeilederOgBruker(
         veilederBrukerKnytninger: List<VeilederBrukerKnytning>,
         tildeltAv: String,
+        token: String,
+        callId: String,
     ) {
-        veilederBrukerKnytninger.forEach {
-            val personident = PersonIdent(it.fnr)
+        veilederBrukerKnytninger.forEach { veilederBrukerKnytning ->
+            val personident = PersonIdent(veilederBrukerKnytning.fnr)
             val personoversiktStatus = personoversiktStatusRepository.getPersonOversiktStatus(personident)
-                ?: PersonOversiktStatus(fnr = it.fnr).also { personOversiktStatus ->
+                ?: PersonOversiktStatus(fnr = veilederBrukerKnytning.fnr).also { personOversiktStatus ->
                     personoversiktStatusRepository.createPersonOversiktStatus(personOversiktStatus)
                 }
 
-            if (personoversiktStatus.enhet == null) {
+            val arbeidstakerEnhet = if (personoversiktStatus.enhet == null) {
                 personBehandlendeEnhetService.updateBehandlendeEnhet(personident)
-            }
-            personoversiktStatusRepository.lagreVeilederForBruker(
-                veilederBrukerKnytning = it,
-                tildeltAv = tildeltAv,
+            } else {
+                personoversiktStatus.enhet
+            } ?: throw IllegalStateException("Enhet for arbeidstaker er null")
+
+            val veiledereForEnhet = veilederClient.getVeiledereForEnhet(
+                callId = callId,
+                enhetId = arbeidstakerEnhet,
+                token = token,
             )
+            if (!veiledereForEnhet.filter { it.enabled }.map { it.ident }.contains(veilederBrukerKnytning.veilederIdent)) {
```

Comment:
Kanskje denne kunne fått en beskrivende variabel? 😊
```suggestion
 	    val isVeilederMemberOfSykmeldtEnhet = !veiledereForEnhet.filter { it.enabled }.map { it.ident }.contains(veilederBrukerKnytning.veilederIdent)
            if (isVeilederMemberOfArbeidstakerEnhet) {
```

Reply:
Fikset nå.